src/fe-text/irssi.c

```diff
@@ -281,7 +281,7 @@ static void check_files(void)
 	struct stat statbuf;
 
 	if (stat(get_irssi_dir(), &statbuf) != 0) {
-		/* ~/.irssi doesn't exist, first time running irssi */
+		/* irssi config folder doesn't exist, first time running irssi */
```

Comment:
according to the comment, the firsttimer depends on the config dir. but for xdg, we want the user to create the xdg-related folder manually. maybe we should change the check to look for the config file? or some other check? what do you think

Reply:
It would solve an edge case.

We only use XDG paths when the user explicitly creates `$XDG_CONFIG_HOME/irssi`. If we had XDG enabled in an irssi run, this check would simply fail. The config directory is already there (as acknowledged by the first statement).

If we are not using the XDG directories, we default to `~/.irssi`. There is a chance that this directory does not exist, if so, it's a firstimer.

If the config file location is specified in the commandline, this would mean that the user is not a firsttimer. It's for this situation where the user might not have `$XDG_CONFIG_HOME` nor `~/.irssi/` (correct me if this is not possible), the code assumes they are a firsttimer. This is not certainly bad, as it can be used to indicate that the config file is specified and a custom directory structure is being used. This also can be subtle and maybe even misleading.

Checking the config file rather than the directory would solve this edge case.